app/services/haystack_report_request.rb

```diff
@@ -0,0 +1,78 @@
+# frozen_string_literal: true
+
+# -- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+# ++
+
+class HaystackReportRequest
+  MAGIC_APPLICATION_NAME = "OpenProject Experimental AI"
+
+  def initialize(user:)
+    @user = user
+  end
+
+  def call(project)
+    response = OpenProject.httpx.post(
+      URI.join("https://webhooks.local", "/print"),
+      json: {
+        project: { id: project.id, type: project.type },
+        user_token:
+      }
+    )
+
+    # TODO: properly parse JSON; expecting something in the format of { "report": { "content": "Lorem ipsum" } }
+    # This should give us some flexibility on adding additional keys to the response if necessary
+    response.dig("report", "content")
```

Comment:
@dominic-braeunlein @shiroginne FYI: This should give you an idea of the request and response structure that I am currently expecting here.

To write down thoughts:
* `user_token` is part of the body, because we are not using it to authenticate the request to haystack at all; if we wanted to do that, we still have the Auth header available for it (checking that OP is allowed to make a request)
* `project.type` is expected to be a string, according to the current enum, the values are: `project`, `program`, `portfolio`
* This code here is not yet dealing with any errors etc; if you have specific error cases in mind that should be handled, please tell me about them